app/tests/test_search_endpoint.py

```diff
@@ -122,176 +267,113 @@ def test_limit_in_post(self):
         response = self.client.post(self.path, data=payload, content_type="application/json")
         self.assertStatusCode(200, response)
         json_data_payload = response.json()
-        self.assertEqual(len(json_data_payload['features']), 1, msg="More than one item returned.")
-
-        # limit in url
-        payload.pop('limit')
-        response = self.client.post(
-            f"{self.path}?limit={limit}", data=payload, content_type="application/json"
-        )
-        self.assertStatusCode(200, response)
-        json_data_url = response.json()
-
-        # compare limit in payload and limit in url
         self.assertEqual(
-            json_data_payload['features'],
-            json_data_url['features'],
-            msg="Limit parameter in payload and in URL yielded different responses."
+            len(json_data_payload['features']), limit, msg="More than one item returned."
```

Comment:
Might be safer to have `limit` as a variable in `msg`, something like: "More than {limit} item(s) returned.", as `limit` is used as a variable in the payload and not hardcoded?